.github/workflows/docs.yml

```diff
@@ -44,4 +45,58 @@ jobs:
         sed -i -e 's/-j auto/-j 1/' Makefile
         make html
         
-        mv build/html/* "${RUNNER_DOCS_DIR}"
+        cp -r build/html "${RUNNER_ARTIFACT_DIR}"
+        
+        # On PRs we also want to upload the docs into our S3 bucket for preview.
+        if [[ ${{ github.event_name == 'pull_request' }} ]]; then
+          cp -r build/html/* "${RUNNER_DOCS_DIR}"
+        fi
+
+  upload:
+    # FIXME: this also needs to happen on created releases
+#    # FIXME: enable this before the PR is merged
+#    if: github.event_name == 'push' && github.event.push.base_ref == 'main'
```

Comment:
As stated, I initially only want to run this on `main` and add the support for releases later on.